### PR TITLE
filter google search links

### PR DESCRIPTION
**Pull request by z3y50n**
This PR extracts the correct url for each link.

image

I couldn't figure a regex to filter out the &amp part so I used find()

### CODE REVIEW

File: core/util/google.py [hunk 1]
```diff
@@ -70,14 +70,15 @@ def run_crawl(self):
 			payload['start'] = set_page(page)
 			if page >= self.limit:
 				break
-		links = self.framework.page_parse(self._pages).findall(r'a href="([^"]+)"')
+		links = self.framework.page_parse(self._pages).findall(r'a href=".*?(https?[^"]+)"')
```
Comment (saeeddhqan) [inline core/util/google.py]: .*? is a dangerous expr. Instead of this, we need a specific regex.
Comment a demo URL so we can discuss.

Comment (z3y50n) [reply core/util/google.py]: `/url?q=https://www.khanacademy.org/computing/computer-programming/programming/intro-to-programming/v/programming-intro&amp;sa=U&amp;ved=2ahUKEwiovqTyp5fvAhWf63MBHaStCCAQuAIwDnoECAQQAg&amp;usg=AOvVaw19d7CnNCKqZQJe8sEy-shq`

This is how the links are returned right now so I wanted to remove the `/url?q=` part

Comment (saeeddhqan) [reply core/util/google.py]: re.sub(r"^/url\?q=", '', link)
for spliting the url, use more stable chars: &amp;ved=..

**Comment by z3y50n** [comment]
I added 2 `re.sub` functions that produce the same result

**Pull request merged**